### PR TITLE
fix: respect different ways to disable analytics

### PR DESCRIPTION
**Pull request by bastiandoetsch**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
This PR adds alternative config parameter names, to support previously communicated ways to disable analytics (e.g. https://docs.snyk.io/snyk-cli/configure-the-snyk-cli/configure-snyk-cli-to-connect-to-snyk-api, https://snyk.io/policies/tracking-and-analytics/)

This is achieved by registering aliases for the `ANALYTICS_DISABLED` configuration key.

```
config.AddAlternativeKeys(configuration.ANALYTICS_DISABLED, []string{"snyk_analytics_disabled", "disable-analytics", "disable_analytics", "snyk_cfg_disable_analytics"})
```

#### Where should the reviewer start?
https://github.com/snyk/cli/blob/277a1e0a1432201ffeed9b3a3eb64e8e791e75c3/cliv2/cmd/cliv2/main.go#L91

#### How should this be manually tested?
use `snyk config set disable-analytics=1` and start snyk with -d. It should NOT output any text in the end regarding sending analytics.

#### Any background context you want to provide?
https://snyksec.atlassian.net/browse/HEAD-325

#### What are the relevant tickets?
https://snyksec.atlassian.net/browse/HEAD-325

### CODE REVIEW

File: cliv2/cmd/cliv2/main.go [hunk 1]
```diff
@@ -88,6 +88,7 @@ func initApplicationConfiguration(config configuration.Configuration) {
 	config.AddAlternativeKeys(configuration.AUTHENTICATION_BEARER_TOKEN, []string{"snyk_oauth_token", "snyk_docker_token"})
 	config.AddAlternativeKeys(configuration.API_URL, []string{"endpoint"})
 	config.AddAlternativeKeys(configuration.ADD_TRUSTED_CA_FILE, []string{"NODE_EXTRA_CA_CERTS"})
+	config.AddAlternativeKeys(configuration.ANALYTICS_DISABLED, []string{"snyk_analytics_disabled", "disable-analytics", "disable_analytics", "snyk_cfg_disable_analytics"})
```
Comment (PeterSchafer) [inline cliv2/cmd/cliv2/main.go:91]: nitpick: the first value is not necessary since the key already is `snyk_disable_analytics`

Comment (bastiandoetsch) [reply cliv2/cmd/cliv2/main.go:91]: but that's internal - it could be anything. In this case, we want to make sure the given options work, independent on what the config key stands for :).

File: test/jest/acceptance/analytics.spec.ts [hunk 2]
```diff
@@ -479,6 +479,63 @@ describe('analytics module', () => {
...
+      SNYK_DISABLE_ANALYTICS: '1',
+    };
+
+    // clear config
+    await runSnykCLI(`config clear`, {
+      env: envWithDisabledAnalytics,
+    });
+
+    // set config
+    await runSnykCLI(`config set disable-analytics=1`, {
```
Comment (PeterSchafer) [inline test/jest/acceptance/analytics.spec.ts:529]: suggestion: this persists the state of analytics being disabled, which might interfere with other tests. Better to undo the change after the test run.

**Review by PeterSchafer — APPROVED** [review]
